### PR TITLE
Add license and Pulumi logo for open source team license

### PR DESCRIPTION
**Pull request by bsweger**
Adding a license and the Pulumi logo as part of applying for an open source team license: https://github.com/pulumi/team-edition-for-open-source?tab=readme-ov-file#how-to-apply

This changeset also does a few pre-committ fixups, including adding a spell check.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,11 +10,15 @@ repos:
...
-    - id: ruff
-      args: [ --fix ]
-    # Run the formatter.
-    - id: ruff-format
+  # Run the linter.
+  - id: ruff
+    args: [ --fix ]
+  # Run the formatter.
+  - id: ruff-format
+- repo: https://github.com/codespell-project/codespell
```
Comment (bsweger) [inline .pre-commit-config.yaml:18]: Will this prevent embarrassing typos in PRs? Only time will tell!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -10,11 +10,15 @@ repos:
 - repo: https://github.com/astral-sh/ruff-pre-commit
   rev: v0.1.11
   hooks:
-    # Run the linter.
```
Comment (bsweger) [inline .pre-commit-config.yaml:13]: :shakes-fist-at-yaml:

File: README.md [hunk 3]
```diff
@@ -26,15 +29,6 @@ Each cloud-enabled hub requires the following AWS resources:
     - A _trust policy_ that stipulates the role can only be used by GitHub Actions that originate from the main branch of the hub's repository.
     - A _permission policy_ that grants write access to the hub's S3 bucket.
 
-This repository contains two Pulumi-related GitHub workflows create and destroy the above AWS resources as needed:
```
Comment (bsweger) [inline README.md:29]: Moved this info down to a dedicated section for "permissions" and removed the OIDC image, which was muddying the waters, imo (we're using OIDC for AWS auth, but it's not the main focus of what the codebase is doing)

**Review by elray1 — APPROVED** [review]
LGTM

**Pull request merged**